cpp-ch/local-engine/Parser/TypeParser.cpp

```diff
@@ -146,19 +146,22 @@ DB::DataTypePtr TypeParser::parseType(const substrait::Type & substrait_type, st
     }
     else if (substrait_type.has_struct_())
     {
-        DB::DataTypes struct_field_types(substrait_type.struct_().types().size());
+        const auto & types = substrait_type.struct_().types();
+        DB::DataTypes struct_field_types(types.size());
         DB::Strings struct_field_names;
-        for (int i = 0; i < static_cast<int>(struct_field_types.size()); ++i)
+        for (int i = 0; i < types.size(); ++i)
         {
+            struct_field_types[i] = parseType(types[i], field_names);
+
```

Comment:
Why move this in front of 'if'

Reply:
Initialize the two var in order as they are passed to `ch_type = std::make_shared<DB::DataTypeTuple>(struct_field_types, struct_field_names);`

Reply:
I am wrong.. it is reverted now.